hottbox/algorithms/decomposition/cpd.py

```diff
@@ -147,7 +150,7 @@ def name(self):
         decomposition_name = super(CPD, self).name
         return decomposition_name
 
-    def decompose(self, tensor, rank, keep_meta=0, kr_reverse=False):
+    def decompose(self, tensor, rank, keep_meta=0, kr_reverse=False, fmat=None):
```

Comment:
As I understand, the main intention behind this change is to reuse this bit within `Parafac2.decompose(fmat=some_llist)` in a form of specific initialisation conditions for CPD-ALS. 

- However, lists are mutable and entries of `fmat` would get changed. It is not critical for the use within `Parafac2.decompose()`, but it could result in undesired effect for the case of `CPD.decompose(fmat=some_list_of_matrices)`. 

- Additially, there is no validation on the shapes of `fmat` within CPD. Thus, incorrect input would result in an Exception coming from `khatri_rao` with a meaningful message. But this could be confusing)

- Double check docstring

Reply:
Checks added